pyrsistent/_pclass.py

```diff
@@ -68,6 +68,10 @@ def __new__(cls, **kwargs):    # Support *args?
 
         check_global_invariants(result, cls._pclass_invariants)
 
+        try:
+            result._hash = hash(tuple((key, getattr(result, key, _MISSING_VALUE)) for key in result._pclass_fields))
+        except:
+           result._hash = None
```

Comment:
Just my 2 cents, but... somewhat concerned about `hasattr()` potential badness since this class is filled with power-features of python.

I think I'd prefer using `None` here and `if self._hash is None` below.

While I in general agree that option types are better, in this case since it's a private member variable, and a pattern everyone has seen before. I don't think the evils of using `None` are sufficiently bad here to avoid it.